packages/repository/src/relations/relation.types.ts

```diff
@@ -108,3 +110,25 @@ export type RelationMetadata =
 
 // Re-export Getter so that users don't have to import from @loopback/context
 export {Getter} from '@loopback/context';
+
+/**
+ * @returns An array of resolved values, the items must be ordered in the same
+ * way as `sourceEntities`. The resolved value can be one of:
+ * - `undefined` when no target model(s) were found
+ * - `Entity` for relations targeting a single model
+ * - `Entity[]` for relations targeting multiple models
+ */
+export type InclusionResolver = (
```

Comment:
Hmm....I couldn't remember is `InclusionResolver` proposed as a type or an interface, the title of PR says it's an interface... 🤔 

As a type, I think it should take a generic type as Source and another type as Return

```ts
export type InclusionResolver<
  S  extends Entity,
  R extends Entity
> = (
  /**
   * List of source models as returned by the first database query.
   */
  sourceEntities: S[],
  /**
   * Inclusion requested by the user (e.g. scope constraints to apply).
   */
  inclusion: Inclusion,
  /**
   * Generic options object, e.g. carrying the Transaction object.
   */
  options?: Options,
  // The return type here is a little bit weird, wouldn't
  // `Promise<R[]>` work?
  // I feel an empty array [] is still a valid value for type `R[]`
) => Promise<(R | undefined)[] | (R[] | undefined)[]>;
```

Thus you don't need to specify the type like https://github.com/strongloop/loopback-next/pull/3517/files#diff-63b076ff3fb794bb126c7d57d33c82bfR323

Reply:
I don't see a need to specify `S` as it does not help the compiler.

Reply:
> Hmm....I couldn't remember is InclusionResolver proposed as a type or an interface, the title of PR says it's an interface... 🤔

It should be a type. We got confused because of the [issue description](https://github.com/strongloop/loopback-next/issues/3445#issue-473228289) where it says interface. We'll fix the commit message/PR title, thanks!

> // The return type here is a little bit weird, wouldn't
>   // `Promise<R[]>` work?
>   // I feel an empty array [] is still a valid value for type `R[]`

So it returns a Promise that has either an array of entities (`(R | undefined)[]`) or an array of an array of entities (`(R[] | undefined)[]`). The former is for something like a `belongsTo` relation where it will return one target entity per source entity. The latter is for something like a `hasMany` relation where it will return multiple target entities per one source entity.

> I don't see a need to specify S as it does not help the compiler.

It helps us with not having to cast in the [case](https://github.com/strongloop/loopback-next/pull/3517/files#diff-63b076ff3fb794bb126c7d57d33c82bfR323) @jannyHou mentioned. I added it in [this commit](https://github.com/strongloop/loopback-next/pull/3517/commits/602aa1bd92af602a72864dacab2c03b0ba41f70b) but if we're fine with casting, I can remove it.